event_routing_backends/processors/mixins/base_transformer.py

```diff
@@ -111,7 +125,52 @@ def extract_username(self):
         Returns:
             str
         """
-        username = self.event.get('context', {}).get('username')
+        username = self.get_data('context.username')
         if not username:
-            username = self.event.get('data', {}).get('username')
+            username = self.get_data('data.username')
         return username
+
+    def get_data(self, key, required=False):
+        """
+        Map the dotted key to nested keys for event dict and return the matching value.
+
+        For example:
+            'key_a.key_b.key_c' will look for the following value:
+
+            {
+                'key_a': {
+                    'key_b': {
+                        'key_c': 'final value'
+                    }
+                }
+            }
+
+        Arguments:
+            key (str)  :    dotted key string for the event dict
+            required (bool) :    key is required or not
+
+        Returns:
+            ANY :                 Returns the value found in the event dict or `None` if
+                                  no value exists for provided dotted path.
+        """
+        result = get_value_from_dotted_path(self.event, key)
+        if result is None:
+            if required:
+                logger.error('Could not get data for %s in event "%s"', key, self.get_data('name', True))
+            else:
+                logger.info('Could not get data for %s in event "%s"', key, self.get_data('name', True))
+
+        return result
+
+    def del_none(self, d):
```

Comment:
arguments should be self explainatory. Instead of `d` we can use `source_dict`

Reply:
Done